libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_RTC/AP_RTC.h>
+#include <GCS_MAVLink/GCS.h>
+#include <GCS_MAVLink/include/mavlink/v2.0/checksum.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define AP_MOUNT_TUSUAV_HEADER1       0x5A     // first header byte
+#define AP_MOUNT_TUSUAV_PACKETLEN_MIN 6        // min number of bytes in a packet (1 header bytes,1 flag,1 cmd,1 sub-cmd,length,1 crc)
+#define AP_MOUNT_TUSUAV_DATALEN_MAX   (AP_MOUNT_TUSUAV_PACKETLEN_MAX-AP_MOUNT_TUSUAV_PACKETLEN_MIN) // max bytes for data portion of packet
+#define AP_MOUNT_TUSUAV_HEALTH_TIMEOUT_MS 1000 // state will become unhealthy if no attitude is received within this timeout
+#define AP_MOUNT_TUSUAV_SEND_INTERVAL_MS 100 // resend angle or rate targets to gimbal at this interval
+#define AP_MOUNT_TUSUAV_PARSE_INTERVAL_MS 1 // resend angle or rate targets to gimbal at this interval
+#define AP_MOUNT_TUSUAV_TIME_SYNC_INTERVAL_MS 1000 // resend angle or rate targets to gimbal at this interval
+#define AP_MOUNT_TUSUAV_ZOOM_SPEED     500    // hard-coded zoom speed (fast)
+#define AP_MOUNT_TUSUAV_ZOOM_MAX       30      // hard-coded absolute zoom times max
+#define AP_MOUNT_TUSUAV_DEBUG 0
+#define debug(fmt, args ...) do { if (AP_MOUNT_TUSUAV_DEBUG) { GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Tusuav: " fmt, ## args); } } while (0)
+
+const char* AP_Mount_Tusuav::send_text_prefix = "Tusuav:";
+
+// update mount position - should be called periodically
+void AP_Mount_Tusuav::update()
+{
+    // exit immediately if not initialised
+    if (!_initialised) {
+        return;
+    }
+    // below here we sent angle or rate targets
+    // throttle sends of target angles or rates
+    uint32_t now_ms = AP_HAL::millis();
+    
+    if (now_ms - _last_update_ms_parse >= AP_MOUNT_TUSUAV_PARSE_INTERVAL_MS) {
+        _last_update_ms_parse = now_ms;
+        // reading incoming packets from gimbal
+        read_incoming_packets();
+    }
+
+    // send time sync
+    if (now_ms - _last_time_sync_ms_send >= AP_MOUNT_TUSUAV_TIME_SYNC_INTERVAL_MS) {
+        _last_time_sync_ms_send = now_ms;
+        send_time_sync();
+    }
+
+    if (now_ms - _last_update_ms_send < AP_MOUNT_TUSUAV_SEND_INTERVAL_MS) {
+        return;
+    }
+    _last_update_ms_send = now_ms;
+
+    //request tracking status and pod angle
+    request_tracking_status();
+
+    // request model name
+    if (!_got_model_name) {
+        request_models();
+    }
+
+    // request firmware version
+    if (!_got_firmware_version) {
+        request_versions();
+    }
+
+    // send vehicle attitude and position
+    send_m_ahrs();
+
+    // change to RC_TARGETING mode if RC input has changed
+    set_rctargeting_on_rcinput_change();
+
+    MountTarget rc_target;
+    get_rc_target(mnt_target.target_type, rc_target);
+	if (mnt_target.target_type == MountTargetType::RATE) {
+		int16_t pitch_value = degrees(rc_target.pitch) / _params.rc_rate_max.get() * 500;
```

Comment:
```suggestion
		const int16_t pitch_value = degrees(rc_target.pitch) / _params.rc_rate_max.get() * 500;
```

etc

Reply:
Thank you for the suggestion.  We have made the modification to use `const` and have reviewed other similar instances for consistency

Reply:
Thank you for the suggestion. We have made the modification to use const and have reviewed other similar instances for consistency.